src/equations/shallow_water_wet_dry_2d.jl

```diff
@@ -0,0 +1,862 @@
+    \frac{\partial}{\partial t}(h v_1) + \frac{\partial}{\partial x}\left(h v_1^2 + \frac{g}{2}h^2\right)
+    + \frac{\partial}{\partial y}(h v_1 v_2) + g h \frac{\partial b}{\partial x} &= 0 \\
+    \frac{\partial}{\partial t}(h v_2) + \frac{\partial}{\partial x}(h v_1 v_2)
+    + \frac{\partial}{\partial y}\left(h v_2^2 + \frac{g}{2}h^2\right) + g h \frac{\partial b}{\partial y} &= 0.
+\end{aligned}
+```
+The unknown quantities of the SWE are the water height ``h`` and the velocities ``\mathbf{v} = (v_1, v_2)^T``.
+The gravitational constant is denoted by `g` and the (possibly) variable bottom topography function ``b(x,y)``.
+Conservative variable water height ``h`` is measured from the bottom topography ``b``, therefore one
+also defines the total water height as ``H = h + b``.
+
+The additional quantity ``H_0`` is also available to store a reference value for the total water height that
+is useful to set initial conditions or test the "lake-at-rest" well-balancedness.
+
+Also, there are two thresholds which prevent numerical problems as well as instabilities. Both of them do not
+have to be passed, as default values are defined within the struct. The first one, `threshold_limiter`, is
+used in [`PositivityPreservingLimiterShallowWater`](@ref) on the water height, as a (small) shift on the initial
+condition and cutoff before the next time step. The second one, `threshold_wet`, is applied on the water height to
+define when the flow is "wet" before calculating the numerical flux.
+
+The bottom topography function ``b(x,y)`` is set inside the initial condition routine
+for a particular problem setup. To test the conservative form of the SWE one can set the bottom topography
+variable `b` to zero.
+
+In addition to the unknowns, TrixiShallowWater.jl currently stores the bottom topography values at the approximation points
+despite being fixed in time. This is done for convenience of computing the bottom topography gradients
+on the fly during the approximation as well as computing auxiliary quantities like the total water height ``H``
+or the entropy variables.
+This affects the implementation and use of these equations in various ways:
+* The flux values corresponding to the bottom topography must be zero.
+* The bottom topography values must be included when defining initial conditions, boundary conditions or
+  source terms.
+* [`AnalysisCallback`](@ref) analyzes this variable.
+* Trixi.jl's visualization tools will visualize the bottom topography by default.
+
+References for the SWE are many but a good introduction is available in Chapter 13 of the book:
+- Randall J. LeVeque (2002)
+  Finite Volume Methods for Hyperbolic Problems
+  [DOI: 10.1017/CBO9780511791253](https://doi.org/10.1017/CBO9780511791253)
+"""
+struct ShallowWaterEquationsWetDry2D{RealT <: Real} <:
+       Trixi.AbstractShallowWaterEquations{2, 4}
+    gravity::RealT # gravitational constant
+    H0::RealT      # constant "lake-at-rest" total water height
```

Comment:
Note that since this could be considered an implementation detail, and thus would be possible to change later. If I were you, I'd merge this PR as-is but put this on the agenda as the very next item (assuming you see merit in my proposition). As usual, it's just a suggestions from my POV, i.e., make sure to verify that it makes sense for you too 🙂

Reply:
I really like this idea, it helps a lot to increase readability of the code. Since the required changes are minor, I have implemented them directly in 033bb482329cf08517bc127cd976613ba2ec4885.